---
title: "rec_rate_vs_repeats_genes_window"
author: "KN"
date: "8/5/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r setup2, echo=FALSE}
#install.packages("ggpubr")

library(ggplot2)
library(ggpubr)
library(viridis)
library("tidyr")
library(plyr)
library(dplyr)
library(data.table)


```

```{r get_data, echo=FALSE, eval=FALSE}
rec_rate <- read.table("tables/rec_rate_markers.table")

#only needed columns
rec_rate_reduced <- rec_rate[,c(1,3,4,5,7,13,14,15,16)]

rec_rate_bin <- as.data.frame(1)
rec_rate_bin <- cbind(rec_rate_bin,rec_rate_reduced[1,])
colnames(rec_rate_bin)[1] <- "Group.1"
rec_rate_bin$rec_sd <- NA

for(i in c(1:length(unique(rec_rate_reduced$map)))) {
  #to get mean per window
  rec_rate_bin_temp <- aggregate(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],], #the data frame
                 by=list(cut(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],"phys"],seq(1, rec_rate_reduced$chr_length[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i]][1]+2000000, 2000000))), #the bins (see below) 
                 mean, na.rm=T ) #the aggregating function
  rec_rate_bin_temp$map <- unique(rec_rate_reduced$map)[i]
  
  rec_rate_bin_temp$rec_sd <- aggregate(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],5], #the data frame
                 by=list(cut(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],"phys"],seq(1, rec_rate_reduced$chr_length[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i]][1]+2000000, 2000000))), #the bins (see below) 
                 sd, na.rm=T ) #the aggregating function
  rec_rate_bin_temp$rec_sd <- rec_rate_bin_temp$rec_sd$x

  #to get nr of markers per window
  rec_rate_bin_temp$mkr <- aggregate(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],1], #the data frame
                 by=list(cut(rec_rate_reduced[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i],"phys"],seq(1, rec_rate_reduced$chr_length[rec_rate_reduced$map==unique(rec_rate_reduced$map)[i]][1]+2000000, 2000000))), #the bins (see below) 
                 length) #the aggregating function
  rec_rate_bin_temp$mkr <- rec_rate_bin_temp$mkr$x
  
  rec_rate_bin <- rbind(rec_rate_bin, rec_rate_bin_temp)
}



str(rec_rate_bin)
head(rec_rate_bin)
rec_rate_bin <- rec_rate_bin[2:length(rec_rate_bin$Group.1),]
#check that nr of markers are the same as in sum stat
aggregate(rec_rate_bin$mkr, by=list(rec_rate_bin$map), sum)

```
```{r manually_change, eval=FALSE, echo=FALSE}

write.table(rec_rate_bin, "tables/rec_rate_bin_2Mb.table")

#manually merge with repeat window analysis in Excel:
# window-based est of repeats
# Rec rate input new lines for W *(rec rate 0, nr of markers 0) and where there are no markers on the map input rec rate NA and 0 nr of markers 
# Remove unnecessary columns
#save as rec_rate_2Mb_windows_vs_repeats_clean.csv

````


```{r get_analysis_data, echo=FALSE}

#read in the merged and cleaned table
rr_repeats_window <- read.csv("tables/rec_rate_2Mb_windows_vs_repeats_clean.csv", sep = ";")
str(rr_repeats_window)
summary(rr_repeats_window)
#rr_repeats_window[,c(4:17)][is.na(rr_repeats_window[,c(4:17)])] <- 0

rr_repeats_window$GC <- read.table("window_based_data/GCA_905220365.1_ilVanCard2.1_genomic_chroms.2MB.GC.tsv")[,5]
  
gene_dens <- read.table("window_based_data/gene_density_2Mb_doublQC.tsv", header = T, fill = TRUE)
colnames(gene_dens)[1] <- "chrom"
#change NA to 0 (count or length are 0)
gene_dens[is.na(gene_dens)] <- 0
str(gene_dens)

gene_dens_CDS <- read.table("window_based_data/gene_CDS_2Mb_doublQC.tsv", header = T, fill = TRUE)
gene_dens_CDS[is.na(gene_dens_CDS)] <- 0
str(gene_dens_CDS)

gene_dens$count_CDS <- gene_dens$count_genes
gene_dens$length_CDS <- gene_dens_CDS$CDS_length


rr_repeats_genes <- join(rr_repeats_window, gene_dens, type = "inner")


rr_repeats_genes$chr_type <- "Autosome"
rr_repeats_genes$chr_type[rr_repeats_genes$chrom=="LR999924.1"] <- "Z"
rr_repeats_genes$chr_type[rr_repeats_genes$chrom=="LR999940.1"] <- "W"


str(rr_repeats_genes)
rr_repeats_genes$length_genes <- as.integer(rr_repeats_genes$length_genes)
rr_repeats_genes$length_CDS <- as.integer(rr_repeats_genes$length_CDS)

#select columns based on string
#select(rr_repeats_window, starts_with("length"))
#or colnames(rr_repeats_genes[, grepl("length_", colnames(rr_repeats_genes))])
#long format
# rr_repeats_genes_long <- pivot_longer(rr_repeats_genes, select(rr_repeats_genes, starts_with("length")), names_to = "repeat_class", values_to = "proportion")
# str(rr_repeats_genes_long)


# rr_repeats_genes_long <- melt(setDT(rr_repeats_genes), measure.vars = list(names(select(rr_repeats_genes, starts_with("count"))),names(select(rr_repeats_genes, starts_with("length")))), variable.name = "gen_feat", value.name = c("count_var", "length_var"))

rr_repeats_genes_long <- melt(setDT(rr_repeats_genes), measure = patterns("count_", "length_"), variable.name = "gen_feat", value.name = c("count_var", "length_var"))

str(rr_repeats_genes_long)
levels(rr_repeats_genes_long$gen_feat) <- names(select(rr_repeats_genes, starts_with("length"))) %>% stringr::str_remove(".+_")
str(rr_repeats_genes_long)

#get density of repeats per kilobase: count*1000/sequence length
rr_repeats_genes_long$density_var <- rr_repeats_genes_long$count_var*1000/(rr_repeats_genes_long$end-rr_repeats_genes_long$begin+1)

#get proportion of genomic features per base: length/sequence length
rr_repeats_genes_long$prop_var <- rr_repeats_genes_long$length_var/(rr_repeats_genes_long$end-rr_repeats_genes_long$begin+1)

#relative position (begin + end)/2 (in the middle of the window)
rr_repeats_genes_long$rel_pos <- ((rr_repeats_genes_long$begin+rr_repeats_genes_long$end)/2)/rr_repeats_genes_long$chr_length
str(rr_repeats_genes_long)

#get wide format for 
rr_repeats_genes_wide <- pivot_wider(rr_repeats_genes_long, names_from = gen_feat, values_from = c(count_var, length_var, density_var, prop_var))
colnames(rr_repeats_genes_wide) <- gsub("_var", "", colnames(rr_repeats_genes_wide))
str(rr_repeats_genes_wide)

```

```{r tables, eval=FALSE, echo=FALSE}

write.table(rr_repeats_genes_long, file="tables/rec_rate_vs_repeats_genes_2Mb_windows_long.table")


#rr_repeats_genes_long <- read.table("tables/rec_rate_vs_repeats_genes_2Mb_windows_long.table")

write.table(rr_repeats_genes_wide, file="tables/rec_rate_vs_repeats_genes_2Mb_windows.table") 


#chromosome level dataframe
repeat_chr_level <- pivot_wider(aggregate(density_var~chrom+gen_feat, rr_repeats_genes_long, mean), names_from = gen_feat, values_from = density_var)
repeat_chr_level_tmp <- pivot_wider(aggregate(density_var~chrom+gen_feat, rr_repeats_genes_long, sd), names_from = gen_feat, values_from = density_var)
repeat_chr_level <- inner_join(repeat_chr_level, repeat_chr_level_tmp, by="chrom", suffix = c("_dens","_dens_sd"))

repeat_chr_level_tmp <- pivot_wider(aggregate(prop_var~chrom+gen_feat, rr_repeats_genes_long, mean), names_from = gen_feat, values_from = prop_var)
repeat_chr_level_tmp2 <- pivot_wider(aggregate(prop_var~chrom+gen_feat, rr_repeats_genes_long, sd), names_from = gen_feat, values_from = prop_var)
repeat_chr_level_tmp <- inner_join(repeat_chr_level_tmp, repeat_chr_level_tmp2, by="chrom", suffix = c("_prop","_prop_sd"))

repeat_chr_level <- inner_join(repeat_chr_level, repeat_chr_level_tmp, by="chrom")

repeat_chr_level$chr_length <- unique(rr_repeats_genes_long$chr_length)
repeat_chr_level$chr_type <- unique(rr_repeats_genes[,c("chrom","chr_type")])$chr_type
repeat_chr_level$rr_mean <- aggregate(rr_repeats_genes_long$rec_rate, by=list(rr_repeats_genes_long$chrom), mean, na.rm=T)$x
repeat_chr_level$rr_sd <- aggregate(rr_repeats_genes_long$rec_rate, by=list(rr_repeats_genes_long$chrom), sd,na.rm=T)$x
repeat_chr_level$GC <- aggregate(rr_repeats_genes_long$GC, by=list(rr_repeats_genes_long$chrom), mean, na.rm=T)$x

write.table(repeat_chr_level, file = "tables/chr_level.table")


#repeat_chr_level <- read.table(file = "tables/chr_level.table")

```



```{r tests}

#function to get the results in a table
#https://stackoverflow.com/questions/34326906/extracting-and-formatting-results-of-cor-test-on-multiple-pairs-of-columns

#run Shapiro-Wilk normality test
#If normaldistr Pearson*s corr otherwise kendall or spearman
#shapiro.test()

corrFunc <- function(var1, var2, data) {
  result = cor.test(data[,var1], data[,var2], method="spearman")
  data.frame(var1, var2, result[c("estimate","p.value","statistic","method")], 
             stringsAsFactors=FALSE)
}



#incl_W
#transfor into dataframe first
COR_TEST <- as.data.frame(rr_repeats_genes_wide)
COR_TEST$density_CDS <- NULL

## Pairs of variables for which we want correlations

#rec_rate
#vars = data.frame(v1=names(mtcars)[1], v2=names(mtcars)[-1])
vars = data.frame(v1=names((COR_TEST)["rec_rate"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
# Apply corrFunc to all rows of vars
corrs_rec = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#tot_repeat_dens
vars = data.frame(v1=names((COR_TEST)["density_rpt"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_tot_repeat_dens = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gen_dens
vars = data.frame(v1=names((COR_TEST)["density_genes"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_gene_dens = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gc
vars = data.frame(v1=names((COR_TEST)["GC"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_gc = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))
#gc
vars = data.frame(v1=names((COR_TEST)["GC"]), v2=names((COR_TEST)["rec_rate"] ))
corrs_gc_rr = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gene_prop vs repeat_prop
vars = data.frame(v1=names((COR_TEST)["prop_CDS"]), v2=names((COR_TEST)["prop_rpt"]))
corrs_prop = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))


corrs_W <- rbind(corrs_rec, corrs_tot_repeat_dens, corrs_gene_dens, corrs_prop, corrs_gc, corrs_gc_rr)



#############################
#without W
COR_TEST <- as.data.frame(rr_repeats_genes_wide[rr_repeats_genes_wide$chr_type!="W",])
COR_TEST$density_CDS <- NULL

#rec_rate
#rec_rate
#vars = data.frame(v1=names(mtcars)[1], v2=names(mtcars)[-1])
vars = data.frame(v1=names((COR_TEST)["rec_rate"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
# Apply corrFunc to all rows of vars
corrs_rec = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#tot_repeat_dens
vars = data.frame(v1=names((COR_TEST)["density_rpt"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_tot_repeat_dens = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gen_dens
vars = data.frame(v1=names((COR_TEST)["density_genes"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_gene_dens = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gc
vars = data.frame(v1=names((COR_TEST)["GC"]), v2=names(select(COR_TEST, starts_with(c("density_","prop_")))))
corrs_gc = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))
#gc
vars = data.frame(v1=names((COR_TEST)["GC"]), v2=names((COR_TEST)["rec_rate"] ))
corrs_gc_rr = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#gene_prop vs repeat_prop
vars = data.frame(v1=names((COR_TEST)["prop_CDS"]), v2=names((COR_TEST)["prop_rpt"]))
corrs_prop = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))


corrs_noW <- rbind(corrs_rec, corrs_tot_repeat_dens, corrs_gene_dens, corrs_prop, corrs_gc, corrs_gc_rr)
colnames(corrs_noW) <- c("var1", "var2", "estimate_noW", "p.value_noW", "statistic_noW", "method")

#join(corrs_W, corrs_noW)
#write.table(join(corrs_W, corrs_noW), "correlations.table")
write.table(join(corrs_W, corrs_noW), "tables/correlations_spearman.table")
#write.table(join(corrs_W, corrs_noW), "correlations_kendall.table")

#cor test versus length/genetic feature is rec rate corr to repeat length?
cor.test(rr_repeats_genes_wide$rec_rate,rr_repeats_genes_wide$length_rpt/rr_repeats_genes_wide$count_rpt)
#t = -3.6812, df = 215, p-value = 0.0002936, coeff -0.2435007 

#without W - non sign
cor.test(rr_repeats_genes_wide[rr_repeats_genes_wide$chr_type!="W",]$rec_rate,rr_repeats_genes_wide[rr_repeats_genes_wide$chr_type!="W",]$length_rpt/rr_repeats_genes_wide[rr_repeats_genes_wide$chr_type!="W",]$count_rpt)
#t = -0.031845, df = 208, p-value = 0.9746, coeff -0.002208071 

######
#Cor with chr length
COR_TEST <- as.data.frame(repeat_chr_level)

## Pairs of variables for which we want correlations

#rec_rate
#vars = data.frame(v1=names(mtcars)[1], v2=names(mtcars)[-1])
vars = data.frame(v1=names((COR_TEST)["chr_length"]), v2=names(select(COR_TEST, ends_with(c("_dens","_prop")))))
# Apply corrFunc to all rows of vars
corrs_length = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

vars = data.frame(v1=names((COR_TEST)["chr_length"]), v2=names(select(COR_TEST, "rr_mean")))
# Apply corrFunc to all rows of vars
corrs_length_rr = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

vars = data.frame(v1=names((COR_TEST)["chr_length"]), v2=names(select(COR_TEST, "GC")))
# Apply corrFunc to all rows of vars
corrs_length_gc = do.call(rbind, mapply(corrFunc, vars[,1], vars[,2], MoreArgs=list(data=COR_TEST), 
                              SIMPLIFY=FALSE))

#corr_length_tot <- rbind(corrs_length_rr, corrs_length)
corr_chr_length <- rbind(corrs_length_rr, corrs_length,corrs_length_gc)

write.table(corr_chr_length, "tables/corr_chr_length.table")

```

```{r corr_heatmap}
#corr_df <-  read.table("correlations.table")
corr_df <-  read.table("tables/correlations_spearman.table")
#corr_df <-  read.table("correlations_kendall.table")


corr_df_long <- melt(setDT(corr_df), measure = patterns("estimate", "p.value", "statistic"), variable.name = c("W"), value.name = c("estimate", "p.value", "statistic"))

levels(corr_df_long$W) <- c("W", "noW")

corr_plot_reduced <- 
ggplot(subset(corr_df_long[var1!="prop_genes",], var2=="density_rpt" | var2=="prop_rpt" | var2=="density_genes" | var2=="prop_CDS"), aes(var1, var2, fill=estimate)) + 
  geom_tile(size=0) +
  facet_wrap(~W) +
  geom_text(aes(label=round(estimate, 2), colour=cut(p.value, c(-Inf, 0.05, Inf)))) +
  scale_fill_gradient2(low = col_rec_rate, high = col_gene) +
  scale_colour_manual(name="p.value", 
                      values = c("(-Inf, 0.05]" = "black",
                                 "(0.05, Inf]" = "light grey")) +
  scale_x_discrete(position = "top") +
    theme(panel.background = element_blank(),
        axis.line = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_text(size = 10, colour = "black"),
        axis.title = element_blank(),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10),
        legend.key = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(size = 12),
        strip.background = element_blank())


corr_plot_rpt_classes <- 
ggplot(corr_df_long, aes(var1, var2, fill=estimate)) + 
  geom_tile(size=0) +
  facet_wrap(~W) +
  geom_text(aes(label=round(estimate, 2), colour=cut(p.value, c(-Inf, 0.05, Inf)))) +
  scale_fill_gradient2(low = "blue", high = "orange") +
  scale_colour_manual(name="p.value", 
                      values = c("(-Inf, 0.05]" = "black",
                                 "(0.05, Inf]" = "light grey")) +
  scale_x_discrete(position = "top") +
    theme(panel.background = element_blank(),
        axis.line = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_text(size = 10, colour = "black"),
        axis.title = element_blank(),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10),
        legend.key = element_blank(),
        strip.placement = "outside", 
        strip.background = element_blank())


ggplot(corr_df, aes(var1, var2, fill=estimate_noW, colour=cut(p.value_noW, c(-Inf, 0.05, Inf)))) + 
  geom_tile(size=2) +
  scale_fill_gradient2(low = col_rec_rate, high = col_gene) +
  scale_colour_manual(name="p.value", 
                      values = c("(-Inf, 0.05]" = "black",
                                 "(0.05, Inf]" = "white")) +
    theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.2, colour = "grey"),
        axis.text = element_text(size = 10),
        axis.title = element_blank(),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 10),
        legend.key = element_blank())
        #legend.position = c(0.8,0.8)) #+

  

```


```{r plots_chr_level}


#library("gridExtra")
#grid.arrange(bxp, dp, bp + rremove("x.text"), 
#             ncol = 2, nrow = 2)
#or
#library(ggpubr)
# figure <- ggarrange(sp, bp + font("x.text", size = 10),
#                     ncol = 1, nrow = 2)
# annotate_figure(figure,
#                 top = text_grob("Visualizing mpg", color = "red", face = "bold", size = 14),
#                 bottom = text_grob("Data source: \n mtcars data set", color = "blue",
#                                    hjust = 1, x = 1, face = "italic", size = 10),
#                 left = text_grob("Figure arranged using ggpubr", color = "green", rot = 90),
#                 right = "I'm done, thanks :-)!",
#                 fig.lab = "Figure 1", fig.lab.face = "bold"
#                 )

#how to run the function: plot_chr_level(VAR)
#assign colour variables
col_Z <- "#009294"   #"#005F60"
col_W <- "#FAAB36"
col_autosome <- "dark grey"

col_rec_rate <- "#249EA0"
col_repeats <- "#FD5900"
col_gene <- "#FAAB36"

#assign variable or dataframe for one plot
VAR <- df$your_variable
#assign colour of regression lines and autosomes
VAR_colour <- "dark grey"

y_axis_title <- "Recombination rate (cM/Mb)" 
x_axis_title <- "Chromosome length (Mb)"

#run with 
plot_chr_level(VAR)

#function definition
plot_chr_level <- function(VAR){
  VAR_mean <- aggregate(VAR, by=list(VAR_df$chrom), mean, na.rm=T)
  VAR_sd <- aggregate(VAR, by=list(VAR_df$chrom), sd, na.rm=T)

  ggplot(VAR_mean, aes(unique(rr_repeats_genes_long$chr_length)/1000000, VAR_mean$x)) +
  geom_smooth(method = "lm", colour=VAR_colour, fill=VAR_colour) +
  geom_pointrange(aes(ymin=VAR_mean$x-VAR_sd$x, ymax=VAR_mean$x+VAR_sd$x), size=0.2) +
  geom_point(aes(fill=unique(rr_repeats_genes_long[,c("chrom","chr_type")])$chr_type), size=3, shape=21) +
  ylab(y_axis_title) +
  xlab(x_axis_title) +
  #scale_colour_discrete(type = c("black", col_W, col_Z)) +
  scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.2, colour = "grey"),
        axis.text = element_text(size = 10),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = c(0.8,0.8),
        axis.ticks = element_line(size = 0.2, colour = "grey")) #+

}


#change this VAR depending on which variable to run
#rec rate
VAR_df <- rr_repeats_genes_long
VAR_colour <- col_autosome
y_axis_title <- "Recombination rate (cM/Mb)" 
x_axis_title <- "Chromosome length (Mb)"

rr_chr_len <- plot_chr_level(VAR_df$rec_rate)

#gc
VAR_df <- rr_repeats_genes_long
VAR_colour <- col_autosome
y_axis_title <- "GC-content (%)" 
x_axis_title <- "Chromosome length (Mb)"

gc_chr_len <- plot_chr_level(VAR_df$GC*100)

#total repeat density
VAR_df <- rr_repeats_genes_long[rr_repeats_genes_long$gen_feat=="rpt",]
VAR_colour <- col_autosome
y_axis_title <- "Repeat density (counts per kb)" 
x_axis_title <- "Chromosome length (Mb)"

rpt_dens_chr_len <- plot_chr_level(VAR_df$density_var)
  
#repeat proportion
VAR_df <- rr_repeats_genes_long[rr_repeats_genes_long$gen_feat=="rpt",]
VAR_colour <- col_autosome
y_axis_title <- "Repeat proportion" 
x_axis_title <- "Chromosome length (Mb)"

rpt_prop_chr_len <- plot_chr_level(VAR_df$prop_var)

#gene density
#repeat density vs chromosome length error bars black
VAR_df <- rr_repeats_genes_long[rr_repeats_genes_long$gen_feat=="genes",]
VAR_colour <- col_autosome
y_axis_title <- "Gene density (counts per kb)" 
x_axis_title <- "Chromosome length (Mb)"

gene_dens_chr_len <- plot_chr_level(VAR_df$density_var)

#gene proportion
VAR_df <- rr_repeats_genes_long[rr_repeats_genes_long$gen_feat=="CDS",]
VAR_colour <- col_autosome
y_axis_title <- "Gene proportion (CDS)" 
x_axis_title <- "Chromosome length (Mb)"

gene_prop_chr_len <- plot_chr_level(VAR_df$prop_var)

#arrange the plots
#remove legend and  x axis title before plotting
figure_chr_len <- ggarrange(rr_chr_len + rremove("x.title"), gc_chr_len + rremove("x.title"), rpt_dens_chr_len + rremove("x.title"), rpt_prop_chr_len + rremove("x.title"), gene_dens_chr_len,gene_prop_chr_len, 
                    labels = c("Figure 1. A", "B", "C", "D", "E", "F"), ncol = 2, nrow = 3,
                    common.legend = TRUE, 
                    legend = "bottom")

# #annotate_figure(figure, 
#                 fig.lab = "Figure 1", 
#                 fig.lab.face = "bold"
#                 )


  
plot_repeat_class_bar <- function(VAR_df,VAR_x,VAR_y,VAR_group) {
  ggplot(VAR_df, aes(VAR_x, VAR_y)) +
    geom_bar(aes(fill=VAR_group, colour=VAR_group), stat = "summary", fun = "mean", position = "dodge") +
    stat_summary(aes(group=VAR_group), fun.data = mean_sd, position = position_dodge(.9), color="black", size=0.2) +
    ylab(y_axis_title) +
    xlab(X_axis_title) +
    scale_colour_discrete(type = c("black", col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "grey"),
          axis.text = element_text(size = 10),
          axis.title.x = element_text(size = 10),
          axis.title.y = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.title = element_blank(),
          legend.key = element_blank())
}



#barplots repeat density per class and chr_type
y_axis_title="Density (counts per kb)"
X_axis_title="Genomic feature"

repeat_cl_bar_dens <- plot_repeat_class_bar(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat, rr_repeats_genes_long$density_var, rr_repeats_genes_long$chr_type)


#barplots repeat prop per class and chr_type
y_axis_title="Proportion"
X_axis_title="Genomic feature"

repeat_cl_bar_prop <- plot_repeat_class_bar(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat, rr_repeats_genes_long$prop_var, rr_repeats_genes_long$chr_type)

#barplots repeat length/repeat count per chr
VAR_df_m=rr_repeats_genes_long
y_axis_title="Mean length per repeat"
X_axis_title="Repeat class"

repeat_cl_bar_mean <- plot_repeat_class_bar(VAR_df_m,VAR_df_m$gen_feat,VAR_df_m$length_var/VAR_df_m$count_var,VAR_df_m$chr_type)

#without genes (tot length)
VAR_df_x=rr_repeats_genes_long[rr_repeats_genes_long$gen_feat!="genes",]
y_axis_title="Mean length per repeat"
X_axis_title="Repeat class"

repeat_cl_bar_mean2 <- plot_repeat_class_bar(VAR_df_x,VAR_df_x$gen_feat,VAR_df_x$length_var/VAR_df_x$count_var,VAR_df_x$chr_type)

figure_repeat_bar_class <- ggarrange(repeat_cl_bar_prop, repeat_cl_bar_dens, repeat_cl_bar_mean,repeat_cl_bar_mean2,
                    labels = c("Figure X. A", "B", "C", "D"), ncol = 2, nrow = 2,
                    common.legend = TRUE, 
                    legend = "bottom")


#boxplot repeat length/repeat count per chr
ggplot(rr_repeats_genes_long) +
  geom_boxplot(aes(chr_type, length_var/count_var, fill=gen_feat)) +
    ylab("Mean length per genomic feature (bases)") +
    scale_fill_discrete() +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "grey"),
          axis.text = element_text(size = 10),
          axis.title.x = element_text(size = 10),
          axis.title.y = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.title = element_blank(),
          legend.key = element_blank())
          #legend.position = c(0.8,0.8)) #+

#boxplot without genes
ggplot(rr_repeats_genes_long[rr_repeats_genes_long$gen_feat!="genes",]) +
  geom_boxplot(aes(gen_feat, log(length_var/count_var), fill=chr_type), outlier.colour = "grey") +
    ylab("Mean length per repeat, log(bases)") +
    scale_fill_discrete(type = c(white, "", col_Z)) +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "grey"),
          axis.text = element_text(size = 10),
          axis.title.x = element_text(size = 10),
          axis.title.y = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.title = element_blank(),
          legend.key = element_blank())



#check length per repeat for the different chromosomes (SINE longer on W)
par(mfrow=c(3,2))
boxplot(rr_repeats_window$length_SINE/rr_repeats_window$count_SINE~rr_repeats_window$chrom)
boxplot(rr_repeats_window$length_DNA/rr_repeats_window$count_DNA~rr_repeats_window$chrom)
boxplot(rr_repeats_window$length_TcM/rr_repeats_window$count_TcM~rr_repeats_window$chrom)
boxplot(rr_repeats_window$length_LTR/rr_repeats_window$count_LTR~rr_repeats_window$chrom)
boxplot(rr_repeats_window$length_nonLTR/rr_repeats_window$count_nonLTR~rr_repeats_window$chrom)


```

```{r plots_regional}

#regional variation in genomic features in 2Mb windows along the chromosome folded (0 is the center, 0.5 is the end of the chromosome)

#function definition
plot_regional_distribution <- function(VAR_df, VAR){
  ggplot(VAR_df, aes(sqrt((rel_pos-0.5)^2), VAR)) +
  geom_point(aes(colour=chr_type, fill=chr_type),size=1, shape=21) +
  geom_smooth(method="loess", colour=col_VAR, fill=col_VAR) +
  scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
  scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
  #facet_wrap(~chrom, scales = "free_x") +
  #scale_x_continuous(breaks = seq(1,max(rr_repeats$begin), 2000000)) +
  ylab(y_axis_title) +
  xlab("Chromosome position") +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.2, colour = "grey"),
        axis.text = element_text(size = 10),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        legend.text = element_text(size = 10),
        axis.ticks = element_line(size = 0.2, colour = "grey"),
        legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "bottom") #+
}




#def variables
#rec rate
y_axis_title <- "Recombination rate (cM/Mb)"
col_VAR="dark grey"
regional_rr <- plot_regional_distribution(rr_repeats_genes_long, rr_repeats_genes_long$rec_rate)

#GC-content
y_axis_title <- "GC-content (%)"
#col_VAR="grey"
regional_gc <- plot_regional_distribution(rr_repeats_genes_long, rr_repeats_genes_long$GC)

#total repeat density
y_axis_title <- "Repeat density (counts per kb)"
#col_VAR=col_repeats
regional_rpt_dens <- plot_regional_distribution(subset(rr_repeats_genes_long, gen_feat=="rpt"), subset(rr_repeats_genes_long, gen_feat=="rpt")$density_var)

#total repeat proportion
y_axis_title <- "Repeat proportion"
#col_VAR=col_repeats
regional_rpt_prop <- plot_regional_distribution(subset(rr_repeats_genes_long, gen_feat=="rpt"), subset(rr_repeats_genes_long, gen_feat=="rpt")$prop_var)

#gene density
y_axis_title <- "Gene density (counts per kb)"
#col_VAR=col_gene
regional_gene_dens <- plot_regional_distribution(subset(rr_repeats_genes_long, gen_feat=="genes"), subset(rr_repeats_genes_long, gen_feat=="genes")$density_var)

#gene proportion
y_axis_title <- "Gene proportion (CDS)"
#col_VAR=col_gene
regional_gene_prop <- plot_regional_distribution(subset(rr_repeats_genes_long, gen_feat=="CDS"), subset(rr_repeats_genes_long, gen_feat=="CDS")$prop_var)



figure_regional_distr <- ggarrange(regional_rr + rremove("x.title"), regional_gc + rremove("x.title"), regional_rpt_dens + rremove("x.title"),regional_rpt_prop + rremove("x.title"), regional_gene_dens, regional_gene_prop,
                    labels = c("Figure X. A", "B", "C", "D","E", "F"), ncol = 2, nrow = 3,
                    common.legend = TRUE, 
                    legend = "bottom")

figure_regional_distr

#checking chromosomal distr
ggplot(rr_repeats_genes_long, aes(rr_repeats_genes_long$rel_pos, rr_repeats_genes_long$GC)) +
  geom_point(aes(colour=chr_type, fill=chr_type),size=1, shape=21) +
  geom_smooth(method="loess", colour="red", fill="red") +
  scale_colour_discrete(type = c(col_VAR, col_W, col_Z)) +
  scale_fill_discrete(type = c("white", col_W, col_Z)) +
  #facet_wrap(~chrom, scales = "free_x") +
  #scale_x_continuous(breaks = seq(1,max(rr_repeats$begin), 2000000)) +
  ylab(y_axis_title) +
  xlab("Chromosome position") +
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.2, colour = "grey"),
        axis.text = element_text(size = 10),
        axis.title.x = element_text(size = 10),
        axis.title.y = element_text(size = 10),
        legend.text = element_text(size = 10),
        legend.title = element_blank(),
        legend.key = element_blank(),
        legend.position = "bottom") #+


```

```{r plots_correlation}
#Linear correlation
repeat_vs_gene_density <- 
ggplot(rr_repeats_genes_wide, aes(density_rpt, density_genes)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_wide, chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    #geom_label(x=0.0012, y=14, label="Cor.coeff=0.341,   \n p-value=4.057e-07", size=3) +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Repeat density (Counts per kb)") +
    ylab("Gene density (Counts per kb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())

repeat_vs_gene_proportion <- 
ggplot(rr_repeats_genes_wide, aes(prop_rpt, prop_CDS)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_wide, chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    #geom_label(x=0.0012, y=14, label="Cor.coeff=0.341,   \n p-value=4.057e-07", size=3) +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Repeat proportion") +
    ylab("Gene proportion (CDS)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())

ggarrange(repeat_vs_gene_density,repeat_vs_gene_proportion, nrow = 2, ncol = 1, common.legend = T, legend = "right")

#rec_rate vs genomic feature
#GC
corr_rr_gc <- 
  ggplot(rr_repeats_genes_long, aes(GC*100, rec_rate)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_long, chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    #facet_wrap(~gen_feat, scales = "free") +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("GC-content (%)") +
    ylab("Recombination rate (cM/Mb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())

corr_rr_rpt <- 
  ggplot(subset(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat=="rpt"), aes(density_var, rec_rate)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat=="rpt" & chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    #facet_wrap(~gen_feat, scales = "free") +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Repeat density (counts per kb)") +
    ylab("Recombination rate (cM/Mb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())

corr_rr_CDS_dens <- 
  ggplot(subset(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat=="CDS"), aes(density_var, rec_rate)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_long, rr_repeats_genes_long$gen_feat=="CDS" & chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    #facet_wrap(~gen_feat, scales = "free") +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Gene density") +
    ylab("Recombination rate (cM/Mb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())


corr_rr_plot <- ggarrange(corr_rr_gc, corr_rr_rpt, corr_rr_CDS_dens, 
          ncol = 1, nrow = 3, 
          common.legend = T, legend = "right")

#multiplots
#density
corr_dens_rr <- 
  ggplot(rr_repeats_genes_long, aes(density_var, rec_rate)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_long, chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    facet_wrap(~gen_feat, scales = "free") +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Density (Counts per kb)") +
    ylab("Recombination rate (cM/Mb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())


#proportion
corr_prop_rr <- 
  ggplot(rr_repeats_genes_long, aes(prop_var, rec_rate)) +
    geom_point(aes(colour=chr_type, fill=chr_type),shape=21,size=1) +
    geom_smooth(method = "lm", colour=col_W, fill=col_W) +
    geom_smooth(data=subset(rr_repeats_genes_long, chr_type != "W"), method = "lm", colour=col_autosome, fill=col_autosome) +
    facet_wrap(~gen_feat, scales = "free") +
    scale_colour_discrete(type = c(col_autosome, col_W, col_Z)) +
    scale_fill_discrete(type = c(col_autosome, col_W, col_Z)) +
    xlab("Proportion") +
    ylab("Recombination rate (cM/Mb)") +
    theme(panel.background = element_blank(),
          axis.line = element_line(size = 0.2, colour = "gray"),
          axis.title = element_text(size = 10),
          legend.text = element_text(size = 10),
          legend.key=element_blank(),
          legend.title = element_blank())


```

```{r lm}
rr_repeats_genes_wide <- as.data.frame(rr_repeats_genes_wide)

rr_repeats_genes_wide_sc <- cbind(rr_repeats_genes_wide[,c("chrom", "chr_type", "rec_rate")], scale(rr_repeats_genes_wide$chr_length), scale(rr_repeats_genes_wide$GC), scale(select(rr_repeats_genes_wide, starts_with("density_"))), scale(select(rr_repeats_genes_wide, starts_with("prop_"))))

colnames(rr_repeats_genes_wide_sc)[4] <- "chr_length"
colnames(rr_repeats_genes_wide_sc)[5] <- "GC"


rr_repeats_genes_wide_sc$rel_pos <- rr_repeats_genes_wide$rel_pos
 

mod1 <- glm(rr_repeats_genes_wide$rec_rate~rr_repeats_genes_wide$chr_length + rr_repeats_genes_wide$density_SINE + rr_repeats_genes_wide$density_DNA + rr_repeats_genes_wide$density_TcM + rr_repeats_genes_wide$density_LINE + rr_repeats_genes_wide$density_LTR + rr_repeats_genes_wide$density_nonLTR + rr_repeats_genes_wide$density_genes)

mod1_sc <- glm(rr_repeats_genes_wide_sc$rec_rate~rr_repeats_genes_wide_sc$chr_length + rr_repeats_genes_wide_sc$density_SINE + rr_repeats_genes_wide_sc$density_DNA + rr_repeats_genes_wide_sc$density_TcM + rr_repeats_genes_wide_sc$density_LINE + rr_repeats_genes_wide_sc$density_LTR + rr_repeats_genes_wide_sc$density_nonLTR + rr_repeats_genes_wide_sc$density_genes)

mod2_sc <- glm(rr_repeats_genes_wide_sc$rec_rate~rr_repeats_genes_wide_sc$chr_length + rr_repeats_genes_wide_sc$chr_length:rr_repeats_genes_wide_sc$chr_type + rr_repeats_genes_wide_sc$density_SINE + rr_repeats_genes_wide_sc$density_DNA + rr_repeats_genes_wide_sc$density_TcM + rr_repeats_genes_wide_sc$density_LINE + rr_repeats_genes_wide_sc$density_LTR + rr_repeats_genes_wide_sc$density_nonLTR + rr_repeats_genes_wide_sc$density_genes)

mod3_sc <- glm(rr_repeats_genes_wide_sc$rec_rate~rr_repeats_genes_wide_sc$chr_length + rr_repeats_genes_wide_sc$chr_type + rr_repeats_genes_wide_sc$rel_pos + rr_repeats_genes_wide_sc$density_SINE + rr_repeats_genes_wide_sc$density_DNA + rr_repeats_genes_wide_sc$density_TcM + rr_repeats_genes_wide_sc$density_LINE + rr_repeats_genes_wide_sc$density_LTR + rr_repeats_genes_wide_sc$density_nonLTR + rr_repeats_genes_wide_sc$density_genes)

#with GC
#glm
mod4_sc <- glm(rr_repeats_genes_wide_sc$rec_rate~rr_repeats_genes_wide_sc$chr_length + rr_repeats_genes_wide_sc$chr_type + rr_repeats_genes_wide_sc$rel_pos + rr_repeats_genes_wide_sc$GC + rr_repeats_genes_wide_sc$density_SINE + rr_repeats_genes_wide_sc$density_DNA + rr_repeats_genes_wide_sc$density_TcM + rr_repeats_genes_wide_sc$density_LINE + rr_repeats_genes_wide_sc$density_LTR + rr_repeats_genes_wide_sc$density_nonLTR + rr_repeats_genes_wide_sc$prop_genes + rr_repeats_genes_wide_sc$density_genes)

#lm
mod4_sc_lm <- lm(rr_repeats_genes_wide_sc$rec_rate~rr_repeats_genes_wide_sc$chr_length + rr_repeats_genes_wide_sc$chr_type + rr_repeats_genes_wide_sc$rel_pos + rr_repeats_genes_wide_sc$GC + rr_repeats_genes_wide_sc$density_SINE + rr_repeats_genes_wide_sc$density_DNA + rr_repeats_genes_wide_sc$density_TcM + rr_repeats_genes_wide_sc$density_LINE + rr_repeats_genes_wide_sc$density_LTR + rr_repeats_genes_wide_sc$density_nonLTR + rr_repeats_genes_wide_sc$prop_genes + rr_repeats_genes_wide_sc$density_genes)


plot(mod2)

#without W
rr_repeats_genes_wide_sc_noW <- rr_repeats_genes_wide_sc[rr_repeats_genes_wide_sc$chr_type!="W",]

library(lme4)


mmod1 <- lmer(rec_rate~chr_length + (1|chr_type) + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc)

mmod1_noW <- lmer(rec_rate~chr_length + (1|chr_type) + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc[rr_repeats_genes_wide$chr_type!="W",])


mmod2 <- lmer(rec_rate~(1|chr_length) + (1|chr_type) + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc)

mmod1_incl_pos <- lmer(rec_rate~chr_length + (1|chr_type) + rel_pos + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc)
summary(mmod1_incl_pos)

#incl GC
mmod1_incl_pos_gc <- lmer(rec_rate~chr_length + (1|chr_type) + rel_pos + GC + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc)
summary(mmod1_incl_pos_gc)

#noW
mmod1_incl_pos_gc_noW <- lmer(rec_rate~chr_length + (1|chr_type) + rel_pos + GC + density_SINE + density_DNA + density_TcM + density_LINE + density_LTR + density_nonLTR + density_genes, rr_repeats_genes_wide_sc_noW)
summary(mmod1_incl_pos_gc_noW)

```

